### PR TITLE
Adds precision to eval

### PR DESCRIPTION
**Pull request by mvpatel2000**
Adds precision to eval. Sets MPT to bf16. For some reason, BF16 + FSDP requires mixed_precision: FULL. It works fine without FSDP. FP16 also works fine and gives basically the same numbers with FSDP on any setting.

### CODE REVIEW

File: scripts/eval/yamls/mpt_eval.yaml
Comment (dakinggg) [inline scripts/eval/yamls/mpt_eval.yaml:4]: should we just use fp32 for all evals? cc @abhi-mosaic

Comment (mvpatel2000) [reply scripts/eval/yamls/mpt_eval.yaml:4]: fp32 does not work with FSDP

**Review by dakinggg — APPROVED** [review]
Please wait for @abhi-mosaic approval as well

**Review by abhi-mosaic — APPROVED** [review]
Looks good for now, and we'll debug `amp_bf16` + FSDP so we can reduce memory+compute requirements in the future.

**Pull request merged**
